bootloader/grub.go

```diff
@@ -446,19 +447,39 @@ func staticCommandLineForGrubAssetEdition(asset string, edition uint) string {
 	return string(cmdline)
 }
 
-var (
-	grubRecoveryModeTrustedAssets = []string{
-		// recovery mode shim EFI binary
-		"EFI/boot/bootx64.efi",
-		// recovery mode grub EFI binary
-		"EFI/boot/grubx64.efi",
-	}
+// bootAssetPath contains the paths for assets in the boot chain.
+type bootAssetPath struct {
+	shimBinary string
+	grubBinary string
+}
 
-	grubRunModeTrustedAssets = []string{
-		// run mode grub EFI binary
-		"EFI/boot/grubx64.efi",
-	}
-)
+// bootAssetsForArch contains the paths for assets for different
+// architectures in a map
+var bootAssetsForArch = map[string]bootAssetPath{
+	"amd64": {
+		shimBinary: filepath.Join("EFI/boot/", "bootx64.efi"),
+		grubBinary: filepath.Join("EFI/boot/", "grubx64.efi")},
+	"arm64": {
+		shimBinary: filepath.Join("EFI/boot/", "bootaa64.efi"),
+		grubBinary: filepath.Join("EFI/boot/", "grubaa64.efi")},
+}
+
+func getBootAssetsForArch() bootAssetPath {
+	return bootAssetsForArch[arch.DpkgArchitecture()]
```

Comment:
It seems with this implementation we need a remark that TrustedAssetsBootloader.TrustedAssets and *Chain  should be used only at runtime/on the device.  It seems to be the case ATM but we should double check. We should error at some level if we need this and PrepareImageTime was true.

Reply:
my suggestion was to capture PrepareImageTime value on the bootloader and return an error from here if PrepareImageTime is true. That will mean turning this and the other helpers into methods of the grub struct.

Reply:
Understood, I've addressed this now